src/index.js

```diff
@@ -28,14 +28,17 @@ export const getSortedObjectPaths = (obj) => {
  *  `babel-plugin-transfor-define` take options of two types: static config and a path to a file that
  *  can define config in any way a user sees fit. getReplacements takes the options and will either
  *  return the static config or get the dynamic config from disk
- * @param  {Object|String}  configOptions  configuration to parse
- * @return {Object}  replacement object
+ * @param  {Object} configOptions  configuration to parse
+ * @return {Object} replacement object
  */
 const getReplacements = (configOptions) => {
-  if (typeof configOptions === "object") { return configOptions; }
+  if (!configOptions || !configOptions.file) {
+    return configOptions;
+  }
 
+  const filePath = configOptions.file;
```

Comment:
What if we updated `getReplacements` to support both a static and dynamic configuration, where dynamic overrides the static.

```js
const getReplacements = (configOptions) => {
  if (!configOptions || !configOptions.filePath) {
    return configOptions;
  }

  const { filePath, ...staticConfig } = configOptions;

  try {
    const fullPath = path.join(process.cwd(), filePath);
    fs.accessSync(fullPath, fs.F_OK);
    const dynamicConfig = require(fullPath);
    return { ...staticConfig, ...dynamicConfig };
  } catch (err) {
    console.error(`The nodePath: ${filePath} is not valid.`); // eslint-disable-line
    throw new Error(err);
  }
};
```

## Example Config

```js
// dynamic-config.js
module.exports = {
  baz: "overrides-static-and-used",
  foo: "bar"
};

// babel.rc
{
  ["transform-define", {
    baz: "overridden-by-dynamic-and-NOT-used",
    filePath: "./dynamic-config.js"
    zip: "zap"
  }],
}

// Results in...
{
  baz: "overrides-static-and-used",
  foo: "bar",
  zip: "zap"
}
```